packages/nextjs/pages/example-ui.tsx

```diff
@@ -1,9 +1,22 @@
 import Head from "next/head";
 import type { NextPage } from "next";
-import { ContractData } from "~~/components/example-ui/ContractData";
-import { ContractInteraction } from "~~/components/example-ui/ContractInteraction";
+import { useScaffoldContractWrite } from "~~/hooks/scaffold-eth";
 
 const ExampleUI: NextPage = () => {
+  const { writeAsync, isLoading, data } = useScaffoldContractWrite({
+    contractName: "YourContract",
+    functionName: "checkin",
+    value: "0.001",
+  });
+
+  console.log("⚡️ ~ file: example-ui.tsx:12 ~ data:", data);
+
+  const { writeAsync: checkoutFunc, isLoading: checkOutLoading } = useScaffoldContractWrite({
+    contractName: "YourContract",
+    functionName: "checkout",
+    deps: [data],
```

Comment:
This feels a bit weird, since the provided dependencies are not actually used internally, so they're not really dependencies. Will play around a bit with it, but it's not clear why the `refetch` mechanism with the `useEffect` is needed.